### DISCUSSION TITLE
SELinux enabled system

### DISCUSSION THREAD

**Discussion by Marc-Harony** [FreshRSS/FreshRSS#4755] [Q&A] [answered]
Hi 👋🏻 
I'm currently running FreshRSS on a SELinux enabled system.
I have made every changes required in #1835, but still, FreshRSS doesn't work properly.

I can correctly login and have access to the main page: ![image](https://user-images.githubusercontent.com/88329570/196877801-40964b04-5c9a-47b7-a6ad-cb82facd84a0.png) (fig1)

But once I try to add new feeds to the app, I get a blank page: 
![image](https://user-images.githubusercontent.com/88329570/196878121-7ff436f6-a160-4768-b4b8-3f815f092ca8.png) (fig2)

I have set SELinux in permissive mode to make the changes but that's not satisfying.

Do you have any ideas on how to solve this issue?
Did I miss something on the installation part and do I have to completely reinstall the app?

Thank you for your help! 🙏🏻  

---

Bonjour 👋🏻 
J'utilises actuellement FreshRSS sur un serveur avec SELinux.
J'ai réalisé toutes les modifications citées dans #1835 mais cela ne semble pas fonctionner.
J'arrive a me connecter à mon compte d'application (fig1) mais quand j'essaie d'ajouter des nouveaux feeds, j'accède à une page vide (fig2)
J'ai temporairement paramétré SELinux en mode permissif, mais ce n'est pas vraiment la meilleure solution.
Ai-je oublié un détail lors de l'installation de FreshRSS ? Dois-je tout réinstaller ?

Merci de l'aide que vous pourrez m'apporter. 👋🏻

**Comment by Alkarex** [comment]
Bonjour,
Essayez de regarder vos logs dans FreshRSS (accessibles depuis le menu configuration) et vos logs du serveur Web.

**Comment by Marc-Harony** [comment, marked as answer]
Hello again 👋🏻 
I have found a way to solve this issue. This is not the most secured way to do it but I think that this is a good balance between easy config and good security.
I have looked at SELinux logs and saw this kind of lines: 
```
SELinux is preventing /usr/sbin/php-fpm from write access on the directory /var/www/FreshRSS/data/users/freshrss/db.sqlite-journal. For complete SELinux messages run: sealert -l 9...

SELinux is preventing /usr/sbin/php-fpm from write access on the file /var/www/FreshRSS/data/users/freshrss/db.sqlite. For complete SELinux messages run: sealert -l b42...

SELinux is preventing /usr/sbin/php-fpm from write access on the file db.sqlite. For complete SELinux messages run: sealert -l b42c...
```
So it seems that the the rule `sudo semanage fcontext -a -t httpd_sys_rw_content_t '/path/to/FreshRSS/data(./*)?'` does not apply recursively.
I found that I had to enter the rule for each subfolder, and each subfolders of subfolders, ...

I have deleted all the rules and set the `http_unified` boolean to true with the following : `sudo setsebool -P httpd_unified 1`

After having set SELinux to enforcing mode and restarted httpd.service the problem was solved!

---

Re-bonjour 👋🏻 
J'ai trouvé un moyen de corriger ce problème. Ce n'est pas la façon la plus sécurisée, mais celle qui, je pense, possède le meilleur rapport facilité de configuration / sécurité.

J'ai jeté un oeil a mes logs SELinux et ces lignes ont attiré mon attention : 
```
SELinux is preventing /usr/sbin/php-fpm from write access on the directory /var/www/FreshRSS/data/users/freshrss/db.sqlite-journal. For complete SELinux messages run: sealert -l 9...

SELinux is preventing /usr/sbin/php-fpm from write access on the file /var/www/FreshRSS/data/users/freshrss/db.sqlite. For complete SELinux messages run: sealert -l b42...

SELinux is preventing /usr/sbin/php-fpm from write access on the file db.sqlite. For complete SELinux messages run: sealert -l b42c...
```
Il semblerait que la règle `sudo semanage fcontext -a -t httpd_sys_rw_content_t '/path/to/FreshRSS/data(./*)?'` ne s'applique pas récursivement. Il fallait en effet que j'applique cette règle aux sous-dossiers, à leur sous-dossiers, etc...

J'ai donc supprimé toutes les règles et j'ai trouvé une autre méthode. J'ai simplement modifié le booléen `http_unified` en true grâce à cette commande : `sudo setsebool -P httpd_unified 1`

Après avoir repassé SELinux en mode "enforcing" et après avoir redémarré le service httpd.service, tout fonctionnait parfaitement !
